package.json

```diff
@@ -51,7 +56,10 @@
     "pretty-quick": "^2.0.1",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
-    "rimraf": "^3.0.2"
+    "rollup": "^2.6.1",
+    "rollup-plugin-babel": "^4.4.0",
+    "rollup-plugin-terser": "^5.3.0",
+    "single-spa-react": "."
```

Comment:
This is to trick webpack into accepting a self-referential import. Ideally we'd do this with the package.json exports field, but it's not yet supported in react. Below is the explanation

We want this:
```js
!(function (e, t) {
  "object" == typeof exports && "undefined" != typeof module
    ? (module.exports = t(require("react"), require("../lib/single-spa-react.js")))
    : "function" == typeof define && define.amd
    ? define(["react", "single-spa-react"], t)
    : ((e = e || self).Parcel = t(e.React, e.singleSpaReact));
})(this, function (e, t) {
  "use strict";
```

But rollup can only give us this (afaik):

```js
!(function (e, t) {
  "object" == typeof exports && "undefined" != typeof module
    ? (module.exports = t(require("react"), require("single-spa-react")))
    : "function" == typeof define && define.amd
    ? define(["react", "single-spa-react"], t)
    : ((e = e || self).Parcel = t(e.React, e.singleSpaReact));
})(this, function (e, t) {
  "use strict";
```

The second one works if `require('single-spa-react')` works as a self referential import. Note that it causes the following yarn warning, but did work for me with yarn install + yarn link:

```
warning Using the "file:" protocol implicitly is deprecated. Please either prepend the protocol or prepend the path "." with "./".
```

Reply:
I tried the following rollup config which I thought would work, but it didn't:

```
  {
    input: "src/parcel.js",
    output: {
      file: "parcel/index.js",
      format: "umd",
      name: "Parcel",
      sourcemap: true,
      globals: {
        react: "React",
        "single-spa-react": "singleSpaReact",
      },
    },
    plugins: [babel(), terser()],
    external: {
      "single-spa-react": "../lib/single-spa-react",
      React: "react",
    },
  },
```